packages/svelte/src/internal/client/dom/blocks/each.js

```diff
@@ -6,36 +6,52 @@ import {
 	EACH_ITEM_REACTIVE,
 	EACH_KEYED
 } from '../../../../constants.js';
-import { noop } from '../../../common.js';
 import {
 	current_hydration_fragment,
 	get_hydration_fragment,
 	hydrate_block_anchor,
 	hydrating,
 	set_current_hydration_fragment
 } from '../hydration.js';
-import { clear_text_content, empty } from '../operations.js';
+import { empty } from '../operations.js';
 import { insert, remove } from '../reconciler.js';
-import { current_block, execute_effect } from '../../runtime.js';
-import { destroy_effect, render_effect } from '../../reactivity/effects.js';
+import { current_block, untrack } from '../../runtime.js';
+import {
+	destroy_effect,
+	pause_effect,
+	render_effect,
+	resume_effect,
+	user_effect
+} from '../../reactivity/effects.js';
 import { source, mutable_source, set } from '../../reactivity/sources.js';
-import { trigger_transitions } from '../elements/transitions.js';
 import { is_array, is_frozen, map_get, map_set } from '../../utils.js';
-import { EACH_BLOCK, EACH_ITEM_BLOCK, STATE_SYMBOL } from '../../constants.js';
+import { STATE_SYMBOL } from '../../constants.js';
+
+var NEW_BLOCK = -1;
+var LIS_BLOCK = -2;
 
-const NEW_BLOCK = -1;
-const MOVED_BLOCK = 99999999;
-const LIS_BLOCK = -2;
+/** @type {import('#client').EachItemBlock | null} */
+export let current_each_item_block = null;
```

Comment:
Would be good to explain here why it's necessary to expose this and get/set this

Reply:
added a comment